0010-release-strategy.adoc

```diff
@@ -0,0 +1,505 @@
+
+ - Need ad-hoc releasing strategy for security bugs
+
+| - More flexibility
+
+- Can focus on meaningful releases
+
+- Less work overhead
+
+| - Decision making process needed to decide when we release (hopefully this document will help with that)
+
+- May fall in a "one more" pattern (let's merge this PR before we release, ok now this PR, etc)
+
+- Need discipline to ensure we _release early, release often_
+
+- Need to not forgot at sprint planning that we are releasing (depending on how much work it means)
+
+|===
+
+=== Recommendation
+
+Considering the current status of our software, the fact that we are pre-mainnet with scarce users, the release process should not be an added burden that creates overhead.
+For this reason, I recommend that we release feature based.
+In <<Release Trigger Guidelines - Recommendation>>, let's review what could be the _rule of thumbs_ of when to release.
+
+== Metacrate version-ing (comit-rs only)
+
+The comit-rs repo contains a number of crates:
+
+- cnd
+- btsieve
+- vendors/*
+
+The Rust dependency graph looks like that:
+
+----
+cnd -> vendors <- btsieve
+----
+
+The functional (due to REST APIs) dependency graph looks like that:
+----
+cnd -> vendors
+  \       ^
+   \      |
+    \-> btsieve
+----
+
+Which means that:
+
+- a breaking REST API change in btsieve involves updating cnd
+- a breaking lib API change in vendors involves updating cnd and/or btsieve
+
+If we were to version every subcrate, then we would need to come up with a strategy (Similar to https://github.com/testcontainers/testcontainers-rs/blob/master/RELEASING.md) to know how:
+
+- What is the semantic version of the sub-crates
+- How is tied to the semantic version of the meta crate comit-rs
+
+Currently comit-rs only make sense in its whole:
+
+- We do not want to consider a user that would only use a subcrate of comit-rs (ie, btsieve) at this stage.
+- If a crate makes sense by itself and should be extracted as its own crate (or contributed to an existing crate), then this should happen first before considering users of such crate
+
+=== Recommendation
+
+Coming up with a meta releasing strategy is added worked with little added value.
+Users do not import the sub-crates and use them (in comparison with test-containers) and only use comit-rs as a binary.
+
+For these reasons, I recommend to keep all sub-crate at 0.1.0 and only increase the meta crate version as a whole.
+
+
+== Release Trigger Guidelines - Recommendation
+
+This is an attempt to consider and review what could be the reason to trigger a release.
+Inline is the author's recommendation
+
+[cols="1,1,1,2"]
+|===
+| Repo (comit-?)| Description | Triggers a release? | Reason
+
+| i/rs | Security bug fix | Yes | We don't want users to lose funds
+| i/rs | Feature that resolves a quaterly KR | Yes | Mark the achievement and consider it done
+| i/rs | Code refactoring | No | Does not bring value to the user
```

Comment:
According to cargo docs (quoted below) this may trigger a release if its a 'breaking change' which a refactor can be in Rust i.e. refactoring enum variants.

> In Rust, breaking changes include adding fields to structs or variants to enums.

Reply:
Yes for libraries but this is a bin,n nobody is supposed to use our enums